lxd/db/containers.go

```diff
@@ -938,7 +940,14 @@ func (c *ClusterTx) ContainerGetSnapshotsFull(project string, name string) ([]Co
 		Type:    int(CTypeSnapshot),
 	}
 
-	return c.ContainerList(filter)
+	snapshots, err := c.ContainerList(filter)
+	if err != nil {
+		return nil, err
+	}
+
+	sort.Slice(snapshots, func(i, j int) bool { return snapshots[i].CreationDate.Before(snapshots[j].CreationDate) })
```

Comment:
@freeekanayaka do we have or would it be easy to add a Sort entry for ContainerFilter so we can have the DB results pre-sorted instead of having to use the `sort` package as @monstermunchkin is doing here?

Reply:
Yes, it is possible. Given we are close to the release, for sake of efficiency I'd suggest to land this branch (which effectively fixes the issue) and then I'll push another PR myself to move from ```sort.Slice``` to ```ORDER BY```, as an optimization. Alternatively, I can work on a patch to apply to this branch and hand it to @monstermunchkin.

Reply:
Sounds good, having the db code do this for us may the up useful in some other spots where we may have non-deterministic output right now.

Reply:
Hm actually reading your reply I have a doubt. Do you wish to have a ContainerFilter parameter for setting a sort-by-creation-date order or do you want to make that ordering a default, without needing to set an explicit parameter?

Reply:
I think configurable would be best. In most cases we want to sort by name but snapshots indeed make more sense chronologically (when querying snapshot content instead of just the URLs).